vimeo-networking/src/main/java/com/vimeo/networking/VimeoClient.java

```diff
@@ -1126,117 +1162,109 @@ public void failure(VimeoError error) {
      *
      * @param uri The Uri that is found in the PictureResource returned from
-     *            {@link #createPictureResource(String, ModelCallback)}
+     *            {@link #createPictureResource(String, VimeoCallback)}
      */
     @Nullable
-    public Call<Object> activatePictureResource(String uri, ModelCallback callback) {
+    public Call<PictureCollection> activatePictureResource(String uri, VimeoCallback<PictureCollection> callback) {
         if (uri == null || uri.trim().isEmpty()) {
             callback.failure(new VimeoError("uri cannot be empty!"));
             return null;
         }
-        HashMap<String, Object> parameters = new HashMap<>();
+        final HashMap<String, Object> parameters = new HashMap<>();
         parameters.put(Vimeo.PARAMETER_ACTIVE, true);
-        Call<Object> call = mVimeoService.edit(getAuthHeader(), uri, parameters);
-        call.enqueue(getRetrofitCallback(callback));
+        final Call<PictureCollection> call = mVimeoService.editPictureCollection(getAuthHeader(), uri, parameters);
+        call.enqueue(callback);
         return call;
     }
-
-    /**
-     * Delete a picture resource
-     * This is helpful if your activation or upload step fails - you can delete the resource allocated
-     *
-     * @param uri The Uri that is found in the PictureResource returned from
-     *            {@link #createPictureResource(String, ModelCallback)}
-     */
-    public void deletePictureResource(String uri, ModelCallback callback) {
-        if (uri == null || uri.trim().isEmpty()) {
-            callback.failure(new VimeoError("uri cannot be empty!"));
-            return;
-        }
-
-        deleteContent(uri, callback);
-    }
-
     // </editor-fold>
 
     // -----------------------------------------------------------------------------------------------------
     // Video actions (Like, Watch Later, Commenting)
     // -----------------------------------------------------------------------------------------------------
     // <editor-fold desc="Video actions (Like, Watch Later, Commenting)">
-    public Call<Object> updateFollow(boolean follow, String uri, ModelCallback callback) {
-        return updateFollow(follow, uri, callback, true);
-    }
-
-    public Call<Object> updateFollow(boolean follow, String uri, ModelCallback callback, boolean enqueue) {
+    public Call updateFollow(boolean follow, String uri, IgnoreResponseVimeoCallback callback) {
         if (follow) {
-            return follow(uri, callback, enqueue);
+            return follow(uri, callback);
         } else {
-            return unfollow(uri, callback, enqueue);
+            return unfollow(uri, callback);
         }
     }
 
-    public Call<Object> follow(String uri, ModelCallback callback, boolean enqueue) {
-        return putContent(uri, null, callback, enqueue);
+    private Call follow(String uri, IgnoreResponseVimeoCallback callback) {
+        return putContent(uri, null, callback);
```

Comment:
This is an interesting approach to cases in which we don't care about the response.